cache-config/t3c-apply/config/config.go

```diff
@@ -533,6 +535,9 @@ If any of the related flags are also set, they override the mode's default behav
 	if tsHome != "" {
 		TSHome = tsHome
 		tsConfigDir = tsHome + "/etc/trafficserver"
+		if cache != nil && *cache == "varnish" {
+			tsConfigDir = tsHome + "/etc/varnish"
```

Comment:
It looks like for varnish cache servers, this will cause it to look for configuration files under `/etc/trafficserver/etc/varnish` (by default I think `tsHome` is `/opt/trafficserver` so that winds up being `/opt/trafficserver/etc/trafficserver/etc/varnish`) - is that really how we want to structure that?

Reply:
I believe the config dir will be either `home + /etc/trafficserver`, `home + /etc/varnish` or if `--trafficserver-home` flag is not used it will default to `/opt/trafficerver/etc/trafficserver` and won't go into the if block (which is not good) other than that it will be `home + /etc/varnish` for varnish case. For example specifying `--trafficserver-home` flag with `/opt/cache` will write the config to `/opt/cache/etc/varnish` and that is what is done in varnish [entrypoint](https://github.com/AbdelrahmanElawady/trafficcontrol/blob/add-parents-varnish/infrastructure/cdn-in-a-box/varnish/run.sh#L85).

I think ultimately default home and config dir shouldn't be related to TS. However, this change will affect CIAB, tests, workflows and some other code depending on that. So, Maybe it would be better if it is done in a separate PR?